File: src/swift/fastinvsqrt.swift
func fastInvSqrt(x: Float) -> Float {
    var i: Int = Int(bitPattern: UInt(x.bitPattern))
    i = 0x5f3759df - (i >> 1)
    let y: Float = Float(bitPattern: UInt32(i))
    return y * (1.5 - 0.5 * x * y * y)
}

while let line = readLine() {
    if let x = Float(line) {
        print(fastInvSqrt(x: x))
    }
}

File: src/javascript/fastinvsqrt.js
require('readline').createInterface({
    input: process.stdin,
}).on('line', (line) => {
    console.log(fastInvSqrt(parseFloat(line)));
});

function fastInvSqrt(x) {
    const i = floatToUInt32(x);
    const j = 0x5f3759df - (i >> 1);
    const y = uint32ToFloat(j);
    return y * (1.5 - 0.5 * x * y * y);
}

function floatToUInt32(x) {
    const buf = Float32Array.of(x).buffer;
    return new Uint32Array(buf)[0];
}

function uint32ToFloat(i) {
    const buf = Uint32Array.of(i).buffer;
    return new Float32Array(buf)[0];
}
